core/impl/src/main/java/pt/webdetails/cdf/dd/model/inst/writer/cdfrunjs/components/amd/CdfRunJsDataSourceComponentWriter.java

```diff
@@ -126,127 +144,80 @@ private void renderCdaDatasource( StringBuilder out,
         dashPath = FilenameUtils.getPath( dashPath );
         cdaPath = RepositoryHelper.normalize( Util.joinPath( dashPath, cdaPath ) );
       }
-      addJsProperty( out, PropertyName.PATH, buildJsStringValue( cdaPath ), INDENT2, false );
+      addJsProperty( out, PropertyName.PATH, buildJsStringValue( cdaPath ), Writer.INDENT2 );
 
     } else {
 
       // legacy
-      addJsProperty(
-          out,
-          PropertyName.SOLUTION,
-          buildJsStringValue( dataSourceComp.tryGetPropertyValue( PropertyName.SOLUTION, "" ) ),
-          INDENT2,
-          false );
-
-      addJsProperty(
-          out,
-          PropertyName.PATH,
-          buildJsStringValue( dataSourceComp.tryGetPropertyValue( PropertyName.PATH, "" ) ),
-          INDENT2,
-          false );
-
-      addJsProperty(
-          out,
-          PropertyName.FILE,
-          buildJsStringValue( dataSourceComp.tryGetPropertyValue( PropertyName.FILE, "" ) ),
-          INDENT2,
-          false );
+      String solution = dataSourceComp.tryGetPropertyValue( PropertyName.SOLUTION, "" );
+      addJsProperty( out, PropertyName.SOLUTION, buildJsStringValue( solution ), Writer.INDENT2 );
+
+      String path = dataSourceComp.tryGetPropertyValue( PropertyName.PATH, "" );
+      addJsProperty( out, PropertyName.PATH, buildJsStringValue( path ), Writer.INDENT2 );
+
+      String file = dataSourceComp.tryGetPropertyValue( PropertyName.FILE, "" );
+      addJsProperty( out, PropertyName.FILE, buildJsStringValue( file ), Writer.INDENT2 );
     }
   }
 
-  private void renderBuiltinCdaDatasource( StringBuilder out,
-                                           CdfRunJsDashboardWriteContext context,
+  private void renderBuiltinCdaDatasource( StringBuilder out, CdfRunJsDashboardWriteContext context,
                                            DataSourceComponent dataSourceComp ) {
 
-    addJsProperty( out, PropertyName.DATA_ACCESS_ID, buildJsStringValue( dataSourceComp.getName() ), INDENT2, true );
+    String dsName = dataSourceComp.getName();
+    addFirstJsProperty( out, PropertyName.DATA_ACCESS_ID, buildJsStringValue( dsName ), Writer.INDENT2 );
 
     String cdeFilePath = context.getDashboard().getSourcePath();
     if ( cdeFilePath != null ) {
-      if ( cdeFilePath.contains( ".wcdf" ) ) {
+      if ( cdeFilePath.contains( ".cdfde" ) ) {
```

Comment:
What about the `else` case? Before it was assumed this was the else...

Reply:
The `.cdfde` replacement wasn't handled in an `else` case. https://github.com/webdetails/cde/blob/master/core/impl/src/main/java/pt/webdetails/cdf/dd/model/inst/writer/cdfrunjs/components/amd/CdfRunJsDataSourceComponentWriter.java#L173

This was a small refactor so that we only need to call `addJsProperty` for `PropertyName.PATH` once.

Reply:
I didn't say it was an else case, but that it behaved like one...